web/source/text/defaultOutput.ts

```diff
@@ -11,68 +11,116 @@ namespace com.keyman.text {
     private constructor() {
     }
 
+    static codeForEvent(Lkc: KeyEvent) {
+      let keyName = Lkc.kName;
+      let n = Lkc.Lcode;
+
+      var code = Codes.keyCodes[keyName];
+      if(!code) {
+        code = n;
+      }
+
+      return code;
+    }
+
+    /**
+     * Serves as a default keycode lookup table.  This may be referenced safely by mnemonic handling without fear of side-effects.
+     * Also used by Processor.defaultRuleOutput to generate output after filtering for special cases.
+     */
+    public static forAny(Lkc: KeyEvent, shiftState: number) {
```

Comment:
As mentioned in the general notes, this is basically the old `defaultKeyOutput` as a lookup-only function.